phpunit/block-supports/layout-test.php

```diff
@@ -642,61 +642,49 @@ public function data_restore_group_inner_container() {
 		switch_theme( 'default' );
 
-		$args_base = array(
-			'block_content' => '<div class="wp-block-group"></div>',
-			'block'         => array(
-				'blockName'    => 'core/group',
-				'innerBlocks'  => array(),
-				'innerHTML'    => '<div class="wp-block-group"></div>',
-				'innerContent' => array(
-					'<div class="wp-block-group"></div>',
-				),
+		$block_content = '<div class="wp-block-group"></div>';
+		$block         = array(
+			'blockName'    => 'core/group',
+			'innerBlocks'  => array(),
+			'innerHTML'    => '<div class="wp-block-group"></div>',
+			'innerContent' => array(
+				'<div class="wp-block-group"></div>',
 			),
+			'attrs'        => $block_attrs,
 		);
 
-		$ids = array_map(
-			function ( $attrs ) use ( $args_base ) {
-				$args = $args_base;
-
-				$args['block']['attrs'] = $attrs;
-
-				/*
-				 * The `appearance-tools` theme support is temporarily added to ensure
-				 * that the block gap support is enabled during rendering, which is
-				 * necessary to compute styles for layouts with block gap values.
-				 */
-				add_theme_support( 'appearance-tools' );
-				$output = gutenberg_render_layout_support_flag( $args['block_content'], $args['block'] );
-				remove_theme_support( 'appearance-tools' );
-
-				/*
-				 * Iterate over the list of classes of the first rendered element to find
-				 * the hash generated for the container's layout class.
-				 */
-				$processor = new WP_HTML_Tag_Processor( $output );
-				$matcher   = '/wp-container-core-group-is-layout-([a-f0-9]{8})/';
-				if ( $processor->next_tag() ) {
-					foreach ( $processor->class_list() as $class ) {
-						if ( preg_match( $matcher, $class, $matches ) ) {
-							return $matches[1];
-						}
-					}
+		/*
+		 * The `appearance-tools` theme support is temporarily added to ensure
+		 * that the block gap support is enabled during rendering, which is
+		 * necessary to compute styles for layouts with block gap values.
+		 */
+		add_theme_support( 'appearance-tools' );
+		$output = gutenberg_render_layout_support_flag( $block_content, $block );
+		remove_theme_support( 'appearance-tools' );
+
+		/*
+		 * Iterate over the list of classes of the first rendered element to find
+		 * the hash generated for the container's layout class.
+		 */
+		$hash      = '';
+		$processor = new WP_HTML_Tag_Processor( $output );
+		$matcher   = '/wp-container-core-group-is-layout-([a-f0-9]{8})/';
+		if ( $processor->next_tag() ) {
+			foreach ( $processor->class_list() as $class ) {
+				if ( preg_match( $matcher, $class, $matches ) ) {
+					$hash = $matches[1];
+					break;
 				}
-				return '';
-			},
-			// Items are repeated to test hash consistency for the same attributes.
-			array( $attrs1, $attrs1, $attrs2, $attrs2 )
-		);
+			}
+		}
 
-		$this->assertSame( $ids[0], $ids[1] );
-		$this->assertSame( $ids[2], $ids[3] );
-		$this->assertNotSame( $ids[0], $ids[2] );
+		$this->assertSame( $expected_hash, $hash );
```

Comment:
Can you assert whether output contains `'wp-container-core-group-is-layout-' . $expected_hash`? That would drastically simplify the implementation and potentially provide a similar level of verification.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/68210/commits/ed07a177b688995c82533832e198bb01dcb0f0f5